### DISCUSSION TITLE
Existing Internal Commands Update Suggestions

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1544] [Suggestions]
I was wondering if any of the following internal command updates would be considered to be added.
I figured I would just combine all into one Suggestion post and update from time to time if I have any new ideas.

`cm_MakeDir`
Add new parameter that allows user the ability to create a new folder automatically (no dialog box) with a predetermined name.
(I am aware that this can be done using an external command, but this could the ability to achieve this for example if using this command in a Lua automation style script)

`cm_PasteFromClipboard`
Think this would be the place to add this.  Add new parameter that allows items that have been Cut or Copied to clipboard and be able to paste them into folder under cursor highlight.  
-Currently only way to do this is you have to enter the folder to paste.
-DC does not currently include a context menu item "Paste In Folder" that some Linux file managers provide.
-Can use mouse to D&D but currently there is no way to set default D&D action to move, requires keyboard modifier to achieve.
-Not sure if this parameter request is possible, but if it is, being that DC is more keyboard centric, this could offer the ability to achieve this action using keyboard only.

Thank You

**Comment by Skif-off** [comment]
> if using this command in a Lua automation style script

[SysUtils.CreateDirectory](https://doublecmd.github.io/doc/en/lua.html#librarysystem)?

**Comment by advice2020** [comment]
Thank you for your response and for providing Lua script to achieve this.
Still feel this would be useful if it is not too difficult to add as DC command itself.
